### DISCUSSION TITLE
how to create plugin from current state

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#827] [Q&A]
I have a plugin in stable state. But I made lots of errors during creation so there is lots of wrong updates. And even worse, before I understand the plugin philosophy I made changes in table structure not in builder. So now I want to move plugin from development into testing - but plugin crashes during installation. Of course I can fix updates (almost 200 updates to check!) but this is plan B. Is there a method to prepare initial table creation based on current database? I know that I won't be able to go backafter this - but it is ok. Let's assume this is fork. No look back :D

**Comment by bennothommo** [comment]
@SledgehammerPL there isn't an automated way of doing this yet. You'd have to hand-craft a migration based on the current table schema.

I believe there is a `create:migration` Artisan command being worked on to make it automatic.

**Comment by SledgehammerPL** [comment]
./artisan create:migration

  Not enough arguments (missing: "plugin, name").

(I'm not sure what is name - my database tables begins with party_structure_ 

I tried: 

./artisan create:migration party structure
create:migration has not been implemented yet

In Inflector.php line 491:

  Doctrine\Inflector\Inflector::singularize(): Argument #1 ($word) must be of type string, null given, called in /var/www/html/winter/vendor/laravel/f
  ramework/src/Illuminate/Support/Pluralizer.php on line 67

**Reply by bennothommo** [reply]
@SledgehammerPL the command isn't ready yet. It will throw an error even if you fill in the command correctly.

**Reply by faselunare** [reply]
my 2 cents

if you have made changes outside the builder i think you need to check all the updates
in the other case, the way i use, 
for each database update
- go trough the update folder 
- open the update_xxxxx_file.php
- copy the line (for example if i add a fields in the table...)
- paste this line in the create_xxxxx_file.php for this table
- remove the update_xxxxx_file.php
- update the version.yaml by removing the line with the update_xxxxxfile.php

this is valid when adding a fields
if otherwise, you remove the field just delete the line in the create_xxxxxxx_file.php
and update the version.yaml file
